week-2-wide-perspective/w02e04-dungeons-and-the-princess/src/rescue-mission.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { describe, it, expect, beforeAll, beforeEach, vi } from "vitest";
+import { Castle } from './model/castle'
+import { Dungeon } from './model/dungeon'
+import { Person } from './model/person'
+import { Underground } from './model/underground'
+import { createRescueMission } from "./helper/mission";
+
+describe('Recuse mission', () => {
+    let princess: Person;
+    let hades: Underground;
+    let cave: Dungeon;
+    let kingdom: Castle;
+
+    let knightJohn: Person
+    let knightBrienne: Person
+    let kingFrancis: Person
+    let peasantMathew: Person
+    let queenBianca: Person
+    let knightBruce: Person
+
+    beforeEach(() => {
+        princess = new Person('Fiona', 'princess')
+        hades = new Underground(princess)
+        cave = new Dungeon(hades)
+        kingdom = new Castle(cave)
+
+        knightJohn = new Person('John', 'knight')
+        knightBrienne = new Person('Brienne', 'knight')
+        kingFrancis = new Person('Francis', 'king')
+        peasantMathew = new Person('Mathew', 'peasant')
+        queenBianca = new Person('Bianca', 'queen')
+        knightBruce = new Person('Bruce', 'knight')
+    })
+
+    it("three knights should rescue the prisoner", () => {
+        const spy = vi.spyOn(princess, 'sayThanks')
+        createRescueMission([knightBruce, knightBrienne, knightJohn], hades, kingdom)
+        expect(spy).toHaveBeenCalledTimes(1);
+    })
+
+
+    it("should throw error if the queen try to enter into the dungeon", () => {
```

Comment:
> _Testy są zwierciadłem Twojego kodu_
> ~ _Paulo Jabiello_

Widać ze szukasz błędu - ale jakiego ?
Zamiast: 
> should throw error if the queen try to enter into the dungeon

Powienieneś robić:
> should throw error FROM the Dungeon if the queen try to enter into it.

W sensie - nie chodzi o to że nazwa testu jest zła :)
- [x] Ty sprawdzasz czy rzuci
- [ ] Nie sprawdzasz skąd jest błąd.

Muisz wiedzieć na podstawie błędu - skąd został rzucony, to część zadania.